### PR TITLE
NXP-32655: add lint & format setup

### PR DESCRIPTION
**Pull request by rakeshkumar1019**
https://jira.nuxeo.com/browse/NXP-32655

```
Linting "admin_console_ui"...

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/auth/services/auth-interceptor.service.ts
Warning:   16:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
Lint warnings found in the listed files.

Warning:   18:27  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/features/home/components/registration-version/registration-version.component.ts
Warning:   [15](https://github.com/nuxeo/nuxeo-admin-console-ui/actions/runs/9413892027/job/25931613760?pr=34#step:7:16):22  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/features/home/store/actions.ts
Warning:   11:18  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/features/home/store/reducers.ts
Warning:   9:10  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/shared/services/persistence.service.spec.ts
Warning:   [18](https://github.com/nuxeo/nuxeo-admin-console-ui/actions/runs/9413892027/job/25931613760?pr=34#step:7:19):37  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/runner/work/nuxeo-admin-console-ui/nuxeo-admin-console-ui/src/app/shared/services/persistence.service.ts
Warning:    7:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
Warning:   15:[21](https://github.com/nuxeo/nuxeo-admin-console-ui/actions/runs/9413892027/job/25931613760?pr=34#step:7:22)  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 8 problems (0 errors, 8 warnings)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
     "start": "ng serve",
     "build": "ng build",
     "watch": "ng build --watch --configuration development",
-    "test": "ng test --code-coverage"
+    "test": "ng test",
```
Comment (swarnadipa-dev) [inline package.json:9]: include --code-coverage

File: package.json [hunk 2]
```diff
@@ -32,20 +34,37 @@
...
     "@types/jasmine": "~4.3.0",
+    "@typescript-eslint/eslint-plugin": "5.62.0",
+    "@typescript-eslint/parser": "5.62.0",
+    "eslint": "^8.51.0",
+    "husky": "^9.0.11",
     "jasmine-core": "~4.6.0",
     "karma": "~6.4.0",
     "karma-chrome-launcher": "~3.2.0",
     "karma-coverage": "~2.2.0",
-    "karma-coverage-istanbul-reporter": "^3.0.3",
```
Comment (swarnadipa-dev) [inline package.json:46]: include this

**Review by swarnadipa-dev — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (rakeshkumar1019) [reply package.json]: added

File: package.json (hunk 2 above)
Comment (rakeshkumar1019) [reply package.json:46]: included

**Review by swarnadipa-dev — APPROVED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Pull request merged**